x-pack/plugins/data_enhanced/server/routes/session.ts

```diff
@@ -166,4 +175,43 @@ export function registerSessionRoutes(router: DataEnhancedPluginRouter, logger:
       }
     }
   );
+
+  router.post(
+    {
+      path: '/internal/session/{id}/_extend',
+      validate: {
+        params: schema.object({
+          id: schema.string(),
+        }),
+      },
+    },
+    async (context, request, res) => {
+      const { id } = request.params;
+      try {
+        const config = await config$.pipe(first()).toPromise();
+        const searchSession = (await context.search.getSession(
+          id
+        )) as SavedObject<SearchSessionSavedObjectAttributes>;
+        const expires = moment(searchSession.attributes.expires).add(
+          config.search.sessions.defaultExpiration
+        );
+        const ttl = `${expires.diff(moment())}ms`;
+        const response = await context.search!.extendSession(id, ttl);
+
+        return res.ok({
+          body: response,
+        });
+      } catch (err) {
+        return res.customError({
```

Comment:
Can you please use `reportServerError`?
This will save time refactoring after merging https://github.com/elastic/kibana/pull/88613